### PR TITLE
NOTICK Fix statemachine slow tests

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

**Comment by lankydan** [comment]
Run tests: slow

**Comment by lankydan** [comment]
Run tests: slow

**Comment by lankydan** [comment]
Run tests: slow

**Comment by lankydan** [comment]
Run tests: slow

File: node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt [hunk 1]
```diff
@@ -407,6 +415,7 @@ class DBCheckpointStorage(
         currentDBSession().save(dbFlowCheckpoint)
         currentDBSession().save(blob)
         currentDBSession().save(metadata)
+        dbFlowException?.let { currentDBSession().save(it) }
```
Comment (kyriathar) [inline node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:418]: Can we please move that to be between `blob` and `metadata` to be aligned with their usual order.

**Review by kyriathar — DISMISSED** [review]
LGTM

**Comment by lankydan** [comment]
Squash with this commit message:

```
NOTICK Save exception for hospitalized session init errors

Save the exception for flows that fail during session init when they are
kept for observation.

Change the exception tidy up logic to only update the flow's status if
the exception was removed.
```

**Pull request merged**